source/common/formatter/stream_info_formatter.cc

```diff
@@ -967,10 +967,19 @@ const StreamInfoFormatterProviderLookupTable& getKnownStreamInfoFormatterProvide
                   });
             }}},
           {"RESPONSE_CODE_DETAILS",
-           {CommandSyntaxChecker::COMMAND_ONLY,
-            [](absl::string_view, absl::optional<size_t>) {
+           {CommandSyntaxChecker::PARAMS_OPTIONAL,
+            [](absl::string_view format, absl::optional<size_t>) {
+              bool allow_whitespaces = (format == "ALLOW_WHITESPACES");
               return std::make_unique<StreamInfoStringFormatterProvider>(
-                  [](const StreamInfo::StreamInfo& stream_info) {
+                  [allow_whitespaces](const StreamInfo::StreamInfo& stream_info) {
+                    if (!allow_whitespaces) {
+                      if (stream_info.responseCodeDetails().has_value() &&
+                          StringUtil::hasEmptySpace(stream_info.responseCodeDetails().value())) {
```

Comment:
Is there a reason you must check that the string has empty spaces instead of immediately calling replaceAllEmptySpace?

Reply:
It was done on purpose purely for performance reason. Calling immediately `replaceAllEmptySpace` requires allocation of a local variable. I assume that in majority of cases there will be no whitespace in `responseCodeDetails` string.

Reply:
I prefer to put the simple branch at the top of the function and quick return. Like

```
if(allow_whitespaces || !stream_info.responseCodeDetails().has_value()) {
  return stream_info.responseCodeDetails();
}

return StringUtil::replaceAllEmptySpace(stream_info.responseCodeDetails().value());
```

Note, we will always create a local varible as the return value. So, it's fine to call the replaceAllEmptySpace directly and the compiler will help us to move it and eliminate unnecessary copy.